Restructure and improve the docs (#679)
dylibso/chicory: docs/docs/index.md

```
-->

> > *Note*: Functions in Wasm can have multiple returns, hence an array, here we're just taking the first returned value.

Reviewer: ```suggestion
> *Note*: Functions in Wasm can return multiple values, hence the array. This function only returns one value, so we take the first value.
```
